dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -926,6 +927,75 @@ void dpnp_rng_rayleigh_c(void* result, const _DataType scale, const size_t size)
     }
 }
 
+template <typename _DataType>
+void dpnp_rng_shuffle_c(
+    void* result, const size_t itemsize, const size_t ndim, const size_t high_dim_size, const size_t size)
+{
+    if (!(size) || !(high_dim_size > 1))
+    {
+        return;
+    }
+
+    char* result1 = reinterpret_cast<char*>(result);
+
+    double* Uvec = nullptr;
+
+    size_t uvec_size = high_dim_size - 1;
+    Uvec = reinterpret_cast<double*>(dpnp_memory_alloc_c(uvec_size * sizeof(double)));
+    mkl_rng::uniform<double> uniform_distribution(0.0, 1.0);
+    auto uniform_event = mkl_rng::generate(uniform_distribution, DPNP_RNG_ENGINE, uvec_size, Uvec);
+    uniform_event.wait();
+
+    if (ndim == 1)
+    {
+        // Fast, statically typed path: shuffle the underlying buffer.
+        // Only for non-empty, 1d objects of class ndarray (subclasses such
+        // as MaskedArrays may not support this approach).
+        // TODO
+        // kernel
+        char* buf = nullptr;
+        buf = reinterpret_cast<char*>(dpnp_memory_alloc_c(itemsize * sizeof(char)));
+        for (size_t i = uvec_size; i > 0; i--)
+        {
+            size_t j = (size_t)(floor((i + 1) * Uvec[i - 1]));
+            memcpy(buf, result1 + j * itemsize, itemsize);
+            memcpy(result1 + j * itemsize, result1 + i * itemsize, itemsize);
+            memcpy(result1 + i * itemsize, buf, itemsize);
+        }
+
+        dpnp_memory_free_c(buf);
+    }
+    else
+    {
+        // Multidimensional ndarrays require a bounce buffer.
+        // TODO
+        // kernel
+        char* buf = nullptr;
+        size_t step_size = (size / high_dim_size) * itemsize; // size in bytes for x[i] element
+        buf = reinterpret_cast<char*>(dpnp_memory_alloc_c(step_size * sizeof(char)));
+        // TODO
+        // nullptr check will be removed after dpnp_memory_alloc_c update
+        if (buf == nullptr)
```

Comment:
no need this check because dpnp_memory_alloc_c rize an exception in this case

Reply:
I have already removed this check from other places of this function after your changes https://github.com/IntelPython/dpnp/commit/e36bb8511ed849976220038ea34506f34ddc5f87 .
Just forget this one.

In anyway I will remove this checks from the current src on refactoring.